src/coreComponents/constitutive/fluid/BlackOilFluidBase.hpp

```diff
@@ -155,10 +155,17 @@ class BlackOilFluidBase : public MultiFluidBase
                             array1d< array1d< real64 > > const & tableValues );
 
   /**
-   * @brief Check that the table values make sense
+   * @brief Check that the formation volume factor table values make sense
    * @param[in] table the values in the oil or gas table
    */
-  void validateTable( TableFunction const & table ) const;
+  virtual void validateFormationVolumeFactorTable( TableFunction const & table ) const;
+
+  /**
+   * @brief Check that the viscosity table values make sense
+   * @param[in] table the values in the oil or gas table
+   */
+  virtual void validateViscosityTable( TableFunction const & table ) const;
+
```

Comment:
Please document that these functions `\throw` an exception in case of error.

Reply:
I added a line in the Doxygen to say that these functions are throwing an exception when the tables are invalid.